src/compiler/checker.ts

```diff
@@ -11882,8 +11882,18 @@ namespace ts {
                 if (symbol.flags & SymbolFlags.Property &&
                     (expr.kind === SyntaxKind.PropertyAccessExpression || expr.kind === SyntaxKind.ElementAccessExpression) &&
                     (expr as PropertyAccessExpression | ElementAccessExpression).expression.kind === SyntaxKind.ThisKeyword) {
-                    const func = getContainingFunction(expr);
-                    return !(func && func.kind === SyntaxKind.Constructor && func.parent === symbol.valueDeclaration.parent);
+                    return !isInConstructor(getContainingFunction(expr));
+                    function isInConstructor(func: FunctionLikeDeclaration) {

```

Comment:
nit: could you renamed the function -> the current name imply me that you are checking if `func` is in constructor. But my understanding is that you are checking if `symbol` is inside the constructor of the class that defines it..`isSymbolReferredInsideOriginalConstructor` or something like that. Others may have better idea for name :sweat_smile: